### PR TITLE
Add v2 test suite

### PR DESCRIPTION
**Pull request by OR13**
This PR adds and experimental test suite for VCDM v2.

Current approach transforms the published schemas, but in the future, when VCDM v2 is a TR, this transformation should not be necessary.

### CODE REVIEW

File: packages/traceability-tests/examples/v2.json [hunk 1]
```diff
@@ -0,0 +1,10509 @@
...
+      "verified": {
+        "protectedHeader": {
+          "iss": "did:jwk:eyJrdHkiOiJPS1AiLCJjcnYiOiJFZDI1NTE5IiwieCI6InpYMnpvR21XMzZUVExfa3czZy1LRlZqaDVJb2FEY29TYzA4aklENHVHcnMifQ",
```
Comment (OR13) [inline packages/traceability-tests/examples/v2.json:13]: @mprorock note the header params.

File: packages/traceability-tests/examples/v2.json [hunk 2]
```diff
@@ -0,0 +1,10509 @@
...
+          "kid": "#0",
+          "alg": "EdDSA",
+          "cty": "vc+ld+json"
```
Comment (OR13) [inline packages/traceability-tests/examples/v2.json:16]: Note the v2 content type.

File: packages/traceability-tests/examples/v2.json [hunk 3]
```diff
@@ -0,0 +1,10509 @@
...
+              ],
+              "streetAddress": "19 Knox St",
+              "addressLocality": "Toronto",
+              "addressRegion": "ON",
+              "addressCountry": "Canada",
+              "postalCode": "M3B 1A2"
+            }
+          },
+          "credentialSchema": {
+            "id": "https://w3id.org/traceability/openapi/components/schemas/credentials/BankAccountCredential.yml",
```
Comment (OR13) [inline packages/traceability-tests/examples/v2.json:59]: Note the self referencing credentialSchema... this is checked... AFTER ... signature verification in the tests.

File: packages/traceability-tests/examples/v2.json [hunk 4]
```diff
@@ -0,0 +1,10509 @@
...
+      "verified": {
+        "protectedHeader": {
+          "iss": "did:jwk:eyJrdHkiOiJPS1AiLCJjcnYiOiJFZDI1NTE5IiwieCI6InpYMnpvR21XMzZUVExfa3czZy1LRlZqaDVJb2FEY29TYzA4aklENHVHcnMifQ",
+          "kid": "#0",
+          "alg": "EdDSA",
+          "cty": "vc+ld+json"
+        },
+        "payload": {
+          "@context": [
+            "https://www.w3.org/ns/credentials/v2",
```
Comment (OR13) [inline packages/traceability-tests/examples/v2.json:20]: v2 context

**Comment by OR13** [comment]
This test suite won't pass, because not all schemas support `credentialSchema` property... given how fundamental to our work this is, it seems like we should correct this.

**Comment by mprorock** [comment]
> This test suite won't pass, because not all schemas support `credentialSchema` property... given how fundamental to our work this is, it seems like we should correct this.

@rhofvendahl might want to take a pass on this to make sure we have proper `credentialSchema` support in place

**Review by mprorock — APPROVED** [review]
@brownoxford I like the approach taken here, particularly with did:jwk instead of did:key for testing - this will cause us to need to update a few things as we shift more to this model

File: packages/traceability-tests/tests/jsonld.test.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  await kyPromise;
+});
+
+it('canonize', async () => {
+  expect(
+    await canonize(
+      {
+        '@context': { '@vocab': 'https://www.w3.org/2018/credentials#' },
+        type: ['VerifiableCredential'],
+        issuer: 'https://example.edu/issuers/565049',
```
Comment (mprorock) [inline packages/traceability-tests/tests/jsonld.test.js:16]:
```suggestion
        issuer: 'https://example.org/issuers/565049',
```

File: packages/traceability-tests/tests/jsonld.test.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      },
+      {
+        format: 'application/n-quads',
+      }
+    )
+  )
+    .toBe(`_:c14n0 <https://www.w3.org/2018/credentials#id> "did:example:ebfeb1f712ebc6f1c276e12ec21" .
+_:c14n1 <https://www.w3.org/2018/credentials#credentialSubject> _:c14n0 .
+_:c14n1 <https://www.w3.org/2018/credentials#issuanceDate> "2010-01-01T19:23:24Z" .
+_:c14n1 <https://www.w3.org/2018/credentials#issuer> "https://example.edu/issuers/565049" .
```
Comment (mprorock) [inline packages/traceability-tests/tests/jsonld.test.js:30]:
```suggestion
_:c14n1 <https://www.w3.org/2018/credentials#issuer> "https://example.org/issuers/565049" .
```

File: packages/traceability-tests/tests/privateKeyJwk.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+const privateKeyJwk = {
+  kty: 'OKP',
+  crv: 'Ed25519',
+  x: 'zX2zoGmW36TTL_kw3g-KFVjh5IoaDcoSc08jID4uGrs',
+  d: '95hNcacuqtWQP70MDYrL6DRUJ8iGg-oq13I0KM5Etw8',
+};
+
+module.exports = privateKeyJwk;
```
Comment (mprorock) [inline packages/traceability-tests/tests/privateKeyJwk.js:8]: This works, but perhaps p256?

Comment (OR13) [reply packages/traceability-tests/tests/privateKeyJwk.js:8]: I can change that part, it just makes the examples longer.

**Review by nissimsan — APPROVED** [review]

File: .github/workflows/v2.test.yml [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+on: [push]
+
+jobs:
+  ci:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Begin CI...
+        uses: actions/checkout@v2
+      - name: Use Node 16
+        uses: actions/setup-node@v1
```
Comment (mkhraisha) [inline .github/workflows/v2.test.yml:12]: use node v3? and checkout v3? also might be a good idea to use cache with it to speed it up.

File: packages/traceability-tests/tests/issue-verify.test.js [hunk 9]
```diff
@@ -0,0 +1,82 @@
+const jose = require('jose');
+const fs = require('fs');
+const { v4: uuidv4 } = require('uuid');
+// pending merge of https://github.com/w3c-ccg/traceability-vocab/pull/735
```
Comment (mkhraisha) [inline packages/traceability-tests/tests/issue-verify.test.js:4]: putting a comment here to make sure this gets changed after pr 735 is merged but before this pr is.

**Review by mkhraisha — APPROVED** [review]

**Comment by OR13** [comment]
Schema validation requires: https://github.com/w3c-ccg/traceability-vocab/pull/747

File: .github/workflows/v2.test.yml (hunk 8 above)
Comment (OR13) [reply .github/workflows/v2.test.yml:12]: Do you have a snippet you would prefer I use here?

**Comment by OR13** [comment]
I should be unblocked, but we might find out more errors in schemas after we are unblocked.

File: .github/workflows/v2.test.yml (hunk 8 above)
Comment (brownoxford) [reply .github/workflows/v2.test.yml:12]: See any of the workflows in trace-interop:
```
...
    steps:
      # Check out repo, set up node, and install dependencies.
      # @see https://github.com/actions/setup-node#usage
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v3
        with:
          node-version: 16.15.1
          cache: 'npm'
      - run: npm ci
...
```

**Comment by OR13** [comment]
Several related issues have been uncovered here...

- https://github.com/w3c-ccg/traceability-vocab/issues/786
- https://github.com/w3c-ccg/traceability-vocab/issues/787

We will probably want to refactor the schema package tests next, so we can start issuing both v1 and v2 credentials as part of CI tests.

**Pull request merged**